### PR TITLE
Add TCF support

### PR DESCRIPTION
**Pull request by jmccrae**
Implements #29

### CODE REVIEW

**Comment by jmccrae** [comment]
@berste - awaiting your review

**Comment by berstearns** [comment]
picking it now to review

File: teanga/corpus.py [hunk 1]
```diff
@@ -687,6 +688,24 @@ def _to_json(self, writer):
...
         json.dump(dct, writer)
 
+    def to_tcf(self, path:str):
+        """Write the corpus to a TCF file.
+
+        Parameters:
+        -----------
+
+        path: str
+            The path to the TCF file.
```
Comment (berstearns) [inline teanga/corpus.py:698]: the docs are not in the mkdocs format,  Parameters:
should be replace by args and "-------" removed. 

        Args:
            path: str
                    The path to the TCF file.

**Comment by berstearns** [comment]
@jmccrae I managed to run a minimal example without much problems.  just a few general comments of what happened to me during reproducing the use case:

- The docstrings are not on the mkdocs formats

- not really important but in my opinion things gets a bit more organized when both the PR and the branch references to issue #29 so when I'm cloning the repo i just need to checkout to "issue-29". I would name the branch with the issue number. 

- There was not a clear example to test the branch, so I had to create a dummy one on my own:
```
from teanga import Corpus, read_tcf
corpus = Corpus()
corpus.add_layer_meta("text")
doc = corpus.add_doc("This is a document.")
corpus.to_tcf("corpus.tcf")
corpus = read_tcf("corpus.tcf")
print([d for d in corpus.docs][0])
#corpus.to_tcf("corpus.tcf")
```
- (line 706 to_tcf is not checking if teangadb exists) When I first tested it without installing teanga.rs, the to_tcf function didn't have a try/except to catch teangadb not existing so I got  a name error,  so maybe we could add an exception error there ?
```
            NameError: name 'teangadb' is not defined
```
-  Then when I tried installing teanga.rs on the read it points to run
```
pip install https://github.com/TeangaNLP/teanga2/releases/download/dev-latest-linux/teanga-0.1.0-cp310-cp310-manylinux_2_34_x86_64.whl
```
which didn't match my python version, so I had to check the releases to install it correctly . So maybe we could improve the teanga.rs README to be more clear on this.

**Comment by jmccrae** [comment]
I have updated the PR.  I will update the teanga.rs repo too

**Pull request merged**